### PR TITLE
Miscellaneous improvements

### PR DESCRIPTION
**Pull request by clebergnu**
A collection of miscellaneous improvements collected "along the way" during other work.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -154,6 +154,7 @@ def get_long_description():
               "Operating System :: POSIX",
               "Topic :: Software Development :: Quality Assurance",
               "Topic :: Software Development :: Testing",
+              "Programming Language :: Python :: 2.7",
```
Comment (apahim) [inline setup.py]: yes, it's important to be clear and specific.

File: avocado/utils/process.py [hunk 2]
```diff
@@ -257,12 +257,18 @@ class CmdResult(object):
...
+    :param exit_status: exit code of the process
+    :type exit_status: int
+    :param stdout: content of the process stdout
+    :type stdout: str
+    :param stderr: content of the process stderr
+    :type stderr: str
+    :param duration: elapsed wall clock time running the process
+    :type duration: float
     :param pid: ID of the process
+    :type pid: int
```
Comment (apahim) [inline avocado/utils/process.py]: ok

File: avocado/core/job.py [hunk 3]
```diff
@@ -329,23 +329,28 @@ def _log_cmdline():
...
     def _get_avocado_git_version():
+        # if running from git sources, there will be a ".git" directory
+        # 3 levels up
+        base_dir = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+        git_dir = os.path.join(base_dir, '.git')
+        if not os.path.isdir(git_dir):
+            return
+        if not os.path.exists(os.path.join(base_dir, 'python-avocado.spec')):
+            return
+
```
Comment (apahim) [inline avocado/core/job.py]: ok, it works as expected.

File: avocado/core/job.py [hunk 4]
```diff
@@ -329,23 +329,28 @@ def _log_cmdline():
...
         try:
             git = path.find_command('git')
         except path.CmdNotFoundError:
             return
-        # We need to get git root as `py2to3` creates this file in BUILD
+
```
Comment (apahim) [inline avocado/core/job.py]: yep

File: setup.py (hunk 1 above)
Comment (lmr) [reply setup.py]: Right, since py3 support is not fully done yet, this looks good.

**Pull request merged**